rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf

```diff
@@ -212,14 +212,14 @@ SecRule REQUEST_METHOD "@rx ^(?:GET|HEAD)$" \
 
 #
 # Require Content-Length or Transfer-Encoding to be provided with
-# every POST request.
+# every POST request if the protocol version is not HTTP/2.
 #
 # -=[ Rule Logic ]=-
-# This chained rule checks if the request method is POST, if so,
-# it checks that a Content-Length or Transfer-Encoding headers are
-# also present.
+# This chained rule checks if the protocol is not HTTP/2, then checks
+# request method is POST, if so, it checks that a Content-Length or
+# Transfer-Encoding headers are also present.
```

Comment:
Not mandatory but it'd be nice to explain why HTTP/2 is excluded here. Quoting the RFC for example.

Reply:
Proposal:

> HTTP/2 does not use the Transfer-Encoding: chunked anymore, because the underlying transport protocol is already using data frames with known length.
> See https://tools.ietf.org/html/rfc7540#section-8.1 which states that the "chunked" encoding must not be used.

Reply:
I'd leave the second part and just quote the rfc as:
```
RFC7540 8.1 p52:
HTTP/2 does not use the Transfer-Encoding: chunked anymore, because the underlying transport protocol is already using data frames with known length.
```

Reply:
Added the quote from RFC.